rosmon_core/src/monitor/node_monitor.cpp

```diff
@@ -591,6 +607,19 @@ void NodeMonitor::gatherCoredump(int signal)
 	ss << "gdb " << m_launchNode->executable() << " " << coreFile;
 
 	m_debuggerCommand = ss.str();
+
+	time_t t = time(nullptr);
+	tm currentTime;
+	memset(&currentTime, 0, sizeof(currentTime));
+	localtime_r(&t, &currentTime);
+
+	char buf[256];
+	strftime(buf, sizeof(buf), "_%Y_%m_%d_%H_%M_%S", &currentTime);
+	std::string core_rename = coreGlob + "_" + m_launchNode->name() + std::string(buf);
+        
+	if (rename(const_cast<char*>(coreGlob.c_str()), const_cast<char*>(core_rename.c_str())) != 0) {
```

Comment:
I think this will break the debugger command in line 609. Lets have a quick chat tomorrow, if you are are not sure of how this is used.